xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-notifiers/xwiki-platform-notifications-notifiers-default/src/main/java/org/xwiki/notifications/notifiers/internal/UserEventManager.java

```diff
@@ -302,4 +315,30 @@ private boolean isEventFiltered(List<NotificationFilter> filters, Event event, D
         }
         return false;
     }
+
+    public void cleanUpFilters(DocumentDeletedEvent event, DocumentReference user)
+    {
+        DocumentReference deletedDocumentReference = event.getDocumentReference();
+        String serializedReference = this.entityReferenceSerializer.serialize(deletedDocumentReference);
+        try {
+            Set<NotificationFilterPreference> matchingPreferences =
+                this.notificationFilterPreferenceManager.getFilterPreferences(user)
```

Comment:
Here it means we won't get only ScopeNotificationFilterPreferences, but could also be other kind of preferences: I guess it makes sense?